tests/test_basic.py

```diff
@@ -344,3 +344,106 @@ def test_rbd_pvc(rook_cluster):
     rook_cluster.kubernetes.kubectl('delete pod csirbd-demo-pod')
     rook_cluster.kubernetes.kubectl('delete pvc rbd-pvc')
     rook_cluster.kubernetes.kubectl('delete sc rook-ceph-block')
+
+
+def set_storage_pattern(rook_cluster, pattern):
+    # create a working-copy of cluster.yaml
+    cluster_working = os.path.join(rook_cluster.ceph_dir, 'cluster_test.yaml')
+    cluster_original = os.path.join(rook_cluster.ceph_dir, 'cluster.yaml')
+
+    cluster_orig_f = open(cluster_original, 'r')
+    cluster_object = yaml.full_load(cluster_orig_f)
+
+    cluster_object['spec']['storage']['deviceFilter'] = pattern
+
+    cluster_work_f = open(cluster_working, 'a')
+    cluster_work_f.write(yaml.dump(cluster_object, Dumper=yaml.Dumper))
+
+    cluster_orig_f.close()
+    cluster_work_f.close()
+
+    return cluster_working
+
+
+def test_storage_pattern(rook_cluster):
+    # cleanup basic installation
+    cluster_original = os.path.join(rook_cluster.ceph_dir, 'cluster.yaml')
+    rook_cluster.kubernetes.kubectl('delete -f ' + cluster_original)
```

Comment:
(just thinking out loud). We could move this test to be the first in `test_basic.py` so that we don't need to install and then remove cluster.yaml. The second test, or subsequent tests, would need to start by ensuring cluster.yaml is set up at some point. Lets leave this optimisation for later though.